com.teamscale.polarion.plugin/src/com/teamscale/polarion/plugin/client/PluginClient.java

```diff
@@ -0,0 +1,237 @@
+            .header("User-Agent", "Teamscale Polarion Plugin")
+            .POST(BodyPublishers.ofString(params))
+            .build();
+    logCookies();
+    System.out.println(
+        "\n[Teamscale Polarion Plugin Client] Attempting to send POST request: " + url);
+    HttpResponse<?> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+    logHeaders(request.headers());
+    System.out.println("\nRESPONSE STATUS CODE: " + response.statusCode());
+    logHeaders(response.headers());
+    logCookies();
+    return response;
+  }
+
+  // TODO: support query parameters
+  private HttpResponse<?> sendGetRequest(String url, String params)
+      throws IOException, InterruptedException {
+    HttpRequest request =
+        HttpRequest.newBuilder(URI.create(url))
+            .header("Accept", "*/*")
+            .header("Accept-Language", "en-US,en;q=0.5")
+            .header("Origin", "teamscale://teamscale.polartion.plugin.client")
+            .header("Sec-Fetch-Dest", "empty")
+            .header("Sec-Fetch-Mode", "cors")
+            .header("Sec-Fetch-Site", "same-origin")
+            .header("User-Agent", "Teamscale Polarion Plugin")
+            .GET()
+            .build();
+    logCookies();
+    System.out.println(
+        "\n[Teamscale Polarion Plugin Client] Attempting to send GET request: " + url);
+    HttpResponse<?> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+    logHeaders(request.headers());
+    System.out.println("\nRESPONSE STATUS CODE: " + response.statusCode());
+    logHeaders(response.headers());
+    logCookies();
+    return response;
+  }
+
+  private void logHeaders(HttpHeaders httpHeaders) {
+    if (httpHeaders != null) {
+      Map<String, List<String>> headersMap = httpHeaders.map();
+      for (Map.Entry<String, List<String>> header : headersMap.entrySet()) {
+        System.out.println(header.getKey() + ":" + String.join("\n", header.getValue()));
+      }
+    }
+  }
+
+  private void logCookies() {
+    List<HttpCookie> cookies =
+        ((CookieManager) CookieHandler.getDefault()).getCookieStore().getCookies();
+    System.out.println("Cookies:");
+    for (HttpCookie cookie : cookies) {
+      System.out.println(cookie.getName() + ":" + cookie.getValue() + " Path=" + cookie.getPath());
+    }
+  }
+
+  public boolean isAuthRequired(@NotNull HttpResponse<?> response) {
+    if (response == null) return false;
+
+    Map<String, List<String>> headersMap = response.headers().map();
+    for (Map.Entry<String, List<String>> header : headersMap.entrySet()) {
+      if (header.getKey().equals("x-com-ibm-team-repository-web-auth-msg")
+          && header.getValue().get(0).equals("authrequired")) return true;
+    }
+    return false;
+  }
+
+  public static void main(String[] args) {
+    try {
+      PluginClient pluginClient = new PluginClient();
+
+      HttpResponse<?> response = pluginClient.postLoginRequest();
```

Comment:
Is the login solemnly handled via cookies? I think within Teamscale we need another possibility to log in.

Reply:
The current Polarion instance we have in prod only has SAML turned on (and login/pwd via SOAP, which is what we currently use in Teamscale). A more recent Polarion version in dev supports login/pwd via http/s which is what I tested (and what yo usee in that client script). And yes, Polarion relies on session cookies and csrfToken as cookie. /j_security_check is their servlet endpoint that authenticates login/pwd (along with cookies that get sent).

Why that kind of authentication wouldn't work on Teamscale? (of course, I'm assuming we would need to write that kind of client code on Teamscale)